### PR TITLE
Resolves_nullreferenceexception_in pipeline for tests

### LINKED ISSUE #31: Tests - NullReferenceException in pipeline

### PR DESCRIPTION
**Pull request by deepend-dev**
Closes #31

- [x] Resolves null reference exception by defining value for object
- [x] Save file with encoding
- [x] Corrects azcontext test
- [x] Remove unnecessary resource name test

### CODE REVIEW

File: Tests/Test-CmAzContext.Tests.ps1 [hunk 1]
```diff
@@ -19,12 +19,12 @@ Describe "*-CmAzContext Lifecycle Tests" {
 		}
 		It "Should fail when Azure is unavailable and -RequireAzure is set" {
 			{
+				Clear-AzContext -Force
 				Set-CmAzContext -ProjectRoot $projectRoot -Environment "Development"
```
Comment (tm-luke) [inline Tests/Test-CmAzContext.Tests.ps1:23]: Shouldn't it be cmazcontext?

File: Tests/Get-CmAzResourceName.Tests.ps1 [hunk 2]
```diff
@@ -43,18 +43,6 @@ Describe "Get-CmAzResourceName" {
...
-    #     $name = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "Core" -Region "UK South" -Name "Keys"
-    #     $name | Should -Be "rg-keys-core-001"
-    #   }
-    #   It "Should not double separators" {
-    #     Set-CmAzContext -Environment "Development" -ProjectRoot $projectRoot
-    #     $name = Get-CmAzResourceName -Resource "Sample2" -Architecture "Core" -Region "UK South" -Name "MyApp"
-    #     $name | Should -Be "default-myapp-dev-uks-001"
-    #   }
-    # }
     Context "Getting correct build ID operations" {
```
Comment (tm-luke) [inline Tests/Get-CmAzResourceName.Tests.ps1:46]: I thought we already removed this?

File: Tests/Test-CmAzContext.Tests.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Tests/Test-CmAzContext.Tests.ps1:23]: No this one is to throw error if Azcontext is not present

File: Tests/Get-CmAzResourceName.Tests.ps1 (hunk 2 above)
Comment (deepend-dev) [reply Tests/Get-CmAzResourceName.Tests.ps1:46]: yes but its in cm-delete branch which is not going in master

**Pull request merged**